### PR TITLE
Document `bucket`

### PR DESCRIPTION
**Pull request by JustForFun88**
Well, I tried to make documentation for the `bucket`, and tried to do it correctly. But maybe I missed something. You can freely edit, add or even throw everything out if  necessary :slightly_smiling_face:.

### CODE REVIEW

File: src/map.rs [hunk 1]
```diff
@@ -7533,6 +7533,7 @@ mod test_map {
     }
 
     #[test]
+    #[allow(clippy::needless_borrow)]
```
Comment (JustForFun88) [inline src/map.rs:7536]: Just because `cargo clippy --all --tests --features serde,rayon,bumpalo -- -D clippy::all` started returning an error

**Review by Amanieu — COMMENTED** [review]
There's quite a few typos, I've corrected some of them but the same typos appear in other comments as well.

**Comment by Amanieu** [comment]
Overall, this looks great! Thanks for helping document the API!

**Comment by JustForFun88** [comment]
Hmm... new clippy lints.  Ok I'll fix it tonight

**Comment by JustForFun88** [comment]
> There's quite a few typos, I've corrected some of them but the same typos appear in other comments as well.

I'm very sorry for the long break. Tried to make changes and corrections. Unfortunately, English is not my native language, so I apologize for possible mistakes and typos. Can you check again?

**Pull request merged**